unmojang/FjordLauncher
# Merge changes from Prism Launcher 9.1

Prism Launcher 9.0 made substantial changes to the authentication logic (https://github.com/PrismLauncher/PrismLauncher/pull/2402), so this merge involved a lot of refactoring and rewriting the authlib-injector authentication code to match Prism Launcher's new structure. But the error handling while adding a new authlib-injector account is now much nicer (no more "Mojang user authentication ended in a network error").

I'd really appreciate volunteers to test this before it's released!

Suggested things to test:

- Microsoft accounts
  + Log in
  + Refresh
  + Restart launcher
  + Launch game
  + Join server
  + Manage skins
- Authlib-injector accounts
  + Log in
  + Refresh
  + Restart launcher
  + Launch game
  + Join server
  + Manage skins (this is ONLY expected to work for Drasl accounts, other authlib-injector servers are not supported yet, see https://github.com/unmojang/FjordLauncher/issues/37)
- FTB Import
- [Prism Launcher's new Java downloader](https://github.com/PrismLauncher/PrismLauncher/pull/2069)
- macOS
- Linux
- Windows
- Anything you can think of
